sdk/storage/Azure.Storage.Common/src/StorageServerTimeout.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// Factory for scopes that apply server timeout to Azure Storage calls.
+    ///
+    /// For more information, see
+    /// <see href="https://docs.microsoft.com/rest/api/storageservices/setting-timeouts-for-blob-service-operations">
+    /// Setting timeouts for Blob service operations</see>,
+    /// <see href="https://docs.microsoft.com/rest/api/storageservices/setting-timeouts-for-file-service-operations">
+    /// Setting timeouts for File service operations</see>,
+    /// <see href="https://docs.microsoft.com/rest/api/storageservices/setting-timeouts-for-queue-service-operations">
+    /// Setting timeouts for Queue service operations</see>.
+    /// </summary>
+    public static class StorageServerTimeout
+    {
+        /// <summary>
+        /// Creates a scope in which all REST requests to Azure Storage would have provided a server timeout.
```

Comment:
```suggestion
        /// Allows you to specify a server timeout for any Storage operations executing on this thread for the duration of the scope.
```

We should also add a sample showing how to do the much cheaper option of creating a cancellation token that times out because that's what most people really want.

Reply:
Yes, but Is this right place? We have one [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md#maximum-execution-time) .

Reply:
I have tweaked the docs a bit and mentioned that they should be using cts and network timeout. the migration guide entry will come later in separate pr.